src/main/java/org/sagebionetworks/bridge/spring/controllers/CRCController.java

```diff
@@ -0,0 +1,381 @@
+        
+        System.out.println(parser.encodeResourceToString(patient));
+    }
+
+    private String findUserId(List<Identifier> identifiers) {
+        if (identifiers != null && !identifiers.isEmpty()) {
+            for (int i=0; i < identifiers.size(); i++) {
+                Identifier id = identifiers.get(i);
+                if (id.getSystem().equals(USER_ID_VALUE_NS)) {
+                    return id.getValue();
+                }
+            }
+        }
+        throw new BadRequestException("Could not find Bridge user ID in identifiers.");
+    }
+
+    private int writeReportAndUpdateState(String userId, JsonNode data, String reportName, AccountStates state) {
+        AccountId accountId = AccountId.forId(APP_ID, userId);
+        Account account = accountService.getAccount(accountId);
+        if (account == null) {
+            throw new EntityNotFoundException(Account.class);
+        }
+        Set<String> callerSubstudyIds = BridgeUtils.getRequestContext().getCallerSubstudies();
+        ReportData report = ReportData.create();
+        report.setDate(JAN1.toString());
+        report.setData(data);
+        report.setSubstudyIds(callerSubstudyIds);
+        
+        DateRangeResourceList<? extends ReportData> results = reportService.getParticipantReport(
+                APP_ID, reportName, account.getHealthCode(), JAN1, JAN2);
+        int status = (results.getItems().isEmpty()) ? 201 : 200;
+        
+        reportService.saveParticipantReport(APP_ID, reportName, account.getHealthCode(), report);
+        
+        updateState(account, state);
+        accountService.updateAccount(account, null);
+        return status;
+    }    
+    
+    private void updateState(StudyParticipant.Builder builder, AccountStates state) {
+        StudyParticipant p = builder.build();
+        Set<String> mutableGroups = new HashSet<>(); 
+        mutableGroups.addAll(p.getDataGroups());
+        mutableGroups.removeAll(ALL_STATES);
+        mutableGroups.add(state.name().toLowerCase());
+        builder.withDataGroups(mutableGroups);
+        
+        Map<String, String> atts = new HashMap<>();
+        atts.putAll(p.getAttributes());
+        atts.put(TIMESTAMP_FIELD, getTimestamp().toString());
+        builder.withAttributes(atts);
+    }
+    
+    private void updateState(Account account, AccountStates state) {
+        Set<String> mutableGroups = new HashSet<>(); 
+        mutableGroups.addAll(account.getDataGroups());
+        mutableGroups.removeAll(ALL_STATES);
+        mutableGroups.add(state.name().toLowerCase());
+        account.setDataGroups(mutableGroups);
+
+        account.getAttributes().put(TIMESTAMP_FIELD, getTimestamp().toString());
+    }
+    
+    // This is bound to one specific account that we've given to Columbia. 
+    // No one else can authenticate for these calls, and the account itself
+    // has no administrative roles so it can't do anything else besides 
+    // calling these methods.
```

Comment:
How do we ensure this works when we test it in local, dev, staging, and prod? Do we need to hack up our database? Would it be better to have this be config driven?

Reply:
Good question! I have changed this so there are two external IDs that are hard-coded to the real study, and to API for testing purposes. In the CRCTest in our integration tests, it creates this user in the api study with the test externalID, and is then able to make these calls. In production we have the production account that will only operate in the cxi-coronavirus study.